test/FunctionsV2/DurableTaskTargetScalerTests.cs

```diff
@@ -132,5 +137,43 @@ public void TestGetScaleMonitor(bool supportsScaleMonitor)
                 Assert.Equal(scaleMonitor, monitor);
             }
         }
+
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async void ScaleHostE2ETest(bool isTbsEnabled)
+        {
+            Action<ScaleOptions> configureScaleOptions = (scaleOptions) =>
+            {
+                scaleOptions.IsTargetScalingEnabled = isTbsEnabled;
+                scaleOptions.MetricsPurgeEnabled = false;
+                scaleOptions.ScaleMetricsMaxAge = TimeSpan.FromMinutes(4);
+                scaleOptions.IsRuntimeScalingEnabled = true;
+                scaleOptions.ScaleMetricsSampleInterval = TimeSpan.FromSeconds(1);
+            };
+            using (FunctionsV2HostWrapper host = (FunctionsV2HostWrapper)TestHelpers.GetJobHost(this.loggerProvider, nameof(this.ScaleHostE2ETest), enableExtendedSessions: false, configureScaleOptions: configureScaleOptions))
+            {
+                await host.StartAsync();
+
+                IScaleStatusProvider scaleManager = host.InnerHost.Services.GetService<IScaleStatusProvider>();
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.FanOutFanIn), 50, this.output);
+                var status = await client.WaitForCompletionAsync(this.output, timeout: TimeSpan.FromSeconds(400));
+                var scaleStatus = await scaleManager.GetScaleStatusAsync(new ScaleStatusContext());
+                await host.StopAsync();
+                Assert.Equal(OrchestrationRuntimeStatus.Completed, status?.RuntimeStatus);
+
+                // We inspect the Host's logs for evidence that the Host is correctly sampling our scaling requests.
+                // the expected logs depend on whether TBS is enabled or not
+                var expectedSubString = "scale monitors to sample";
+                if (isTbsEnabled)
+                {
+                    expectedSubString = "target scalers to sample";
+                }
+
+                bool containsExpectedLog = this.loggerProvider.GetAllLogMessages().Select(p => p.FormattedMessage ?? "").Any(p => p.Contains(expectedSubString));
+                Assert.True(containsExpectedLog);
+            }
+        }
```

Comment:
@alrod: an E2E test, using the pre-existing DF testing infra, has been added here ^

Reply:
I realize this is not exactly like the E2E tests you suggested (doing so would have taken more time due to how this codebase is structured and the DF initialization boilerplate), but I think this, in addition to the unit tests we already have, should have sufficient testing coverage.